Cython/Compiler/Code.py

```diff
@@ -1411,6 +1413,15 @@ def get_py_string_const(self, text, identifier=None,
             text.encoding, identifier, is_str, py3str_cstring)
         return py_string
 
+    def get_py_codeobj_const_cname(self):
+        idx = len(self.codeobject_constants)
+        name = f"{Naming.codeobjtab_cname}[{idx}]"
+        self.codeobject_constants[name] = None
+        return name
```

Comment:
Why not pass the `CodeObjectNode` as argument and build an ordered list of those nodes here? Then the `ModuleNode` could simply call a dedicated code generation method on each of those CO nodes and the whole CO table filling would be done by the `CodeObjectNode` implementation, rather than being split across two nodes in two different files that need to exchange structured data to do their work correctly.